perun-core/src/test/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntryIntegrationTest.java

```diff
@@ -404,6 +404,40 @@ public void addFacilityToConsentHubAgain() throws Exception {
 		assertThatExceptionOfType(FacilityAlreadyAssigned.class).isThrownBy(() -> perun.getConsentsManagerBl().addFacility(sess, hub, facility));
 	}
 
+	@Test
+	public void changeConsentStatus() throws Exception {
+		System.out.println(CLASS_NAME + "changeConsentStatus");
+
+		ConsentStatus status = ConsentStatus.GRANTED;
+		Facility facility = setUpFacility();
+		User user = setUpUser("John", "Doe");
+
+		Consent consent = new Consent(-1, user.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility.getName()), new ArrayList<>());
+		perun.getConsentsManagerBl().createConsent(sess, consent);
+
+		Consent updatedConsent = consentsManagerEntry.changeConsentStatus(sess, consent, status);
+		assertEquals("Updated consent has different status than sent consent.", status, updatedConsent.getStatus());
```

Comment:
Just a detail, I would probably assert that `getConsentById` returns the consent with GRANTED status as well. Just to be sure that the status was really set in DB, not just to the object returned by `changeConsentStatus`.

Reply:
No problem, I can add one more assert, but changeStatus already returns consent from the database by using getConsentById. Should I return consent in bl layer with set status passed like argument instead of one from the database?

Reply:
I think both ways are alright, you don't have to change it.